git/operations.go

```diff
@@ -42,6 +43,21 @@ func clone(ctx context.Context, workingDir string, keyRing ssh.KeyRing, repoURL,
 }
 
 func commit(ctx context.Context, workingDir, commitMessage string) error {
+	user := ctx.Value("user").(string)
+
+	// The user in the update.Cause.User must be a valid github user, otherwise the commit fails
```

Comment:
Not quite accurate. `git` (not github) will reject a commit if you supply _only_ an email address rather than e.g., `"Michael Bridgen <michael@weave.works>"`, and that email address has not been used in a prior revision.

This is certainly a wrinkle though. Some alternatives:

 - assume that commits will have been made by a particular user, and let git reject those that aren't
 - make people supply a git-author-like value as the username, e.g., `fluxctl release --user="Michael Bridgen <michael@weave.works>" ...`. Not so easy if you are the web UI and you simply don't have all of that information.
 - let people toggle this behaviour, and then do one of the above in the case that it's switched on